File: clojure/1.clj
(defn multiple-3-or-5? [a] (or (= 0 (mod a 3)) (= 0 (mod a 5))))

(defn sum-multiples [limit]
  (let [number-range (range 1 limit)
        multiples (filter multiple-3-or-5? number-range)]
    (reduce + multiples)))

(print (sum-multiples 1000))

File: elixir/1.ex
defmodule Multiple do
  def of?(x, n) do
    (rem n, x) == 0
  end
  def of_3_or_5?(n) do
    (of? 3, n) or (of? 5, n)
  end
end

limit = 1000 - 1
multiples = Enum.filter 1..limit, &(Multiple.of_3_or_5?/1)
sum = Enum.reduce multiples, 0, &(&1 + &2)
IO.puts sum
